### PR TITLE
[ARITH] Add Lowering rule for FloorDiv/Mod

### PR DESCRIPTION
**Pull request by tqchen**
This serves as a step to migrate integer analysis to make use of floormod/div

cc @icemelon9 @yzhliu @sgrechanik-h @umangyadav

### CODE REVIEW

File: src/arithmetic/rewrite_simplify.cc [hunk 1]
```diff
@@ -1674,6 +1674,14 @@ Mutate_(const Call* op, const Expr& self) {
   if (op == nullptr) return ret;
   if (op->is_intrinsic(Call::likely) && is_const(op->args[0])) {
     return op->args[0];
+  } else if (op->is_intrinsic(Call::shift_right)) {
+    if (op->args[0].as<IntImm>() && op->args[1].as<IntImm>()) {
+      return op->args[0] >> op->args[1];
```
Comment (umangyadav) [inline src/arithmetic/rewrite_simplify.cc]: Do we have `shift_right` defined on the `IntImm` ? Should it be `op->args[0]->value >> op->args[1]->value` ? 
Same for the `and`.

Comment (tqchen) [reply src/arithmetic/rewrite_simplify.cc]: the construction itself will constant fold, i will add a followup comment.

Comment (umangyadav) [reply src/arithmetic/rewrite_simplify.cc]: Oh, I see. Okay, thanks.

**Comment by tqchen** [comment]
Thanks @umangyadav

**Pull request merged**